framework/sysdeps/windows/tmpfile.c

```diff
@@ -43,19 +43,30 @@ static unsigned next_random()
 int open_memfd(enum MemfdCloexecFlag flag)
 {
     wchar_t tmpname[sizeof SANDSTONE_STRINGIFY(UINT_MAX) ".tmp"];
+    wchar_t tmppath[MAX_PATH];
+
     HANDLE hFile = INVALID_HANDLE_VALUE;
     SECURITY_ATTRIBUTES sa = {};
     sa.nLength = sizeof(sa);
     sa.bInheritHandle = (flag == MemfdInheritOnExec);
     sa.lpSecurityDescriptor = NULL;
 
+    DWORD getTempPathRes = GetTempPathW(MAX_PATH, tmppath);
+
+    if (getTempPathRes == 0)
+        return -1;
+
     for (int i = 0; hFile == INVALID_HANDLE_VALUE && i < 16; ++i) {
         wcscat(_ultow(next_random(), tmpname, 36), L".tmp");            // yes, base 36
         DWORD access = GENERIC_READ | GENERIC_WRITE;
         DWORD sharemode = FILE_SHARE_DELETE | FILE_SHARE_READ | FILE_SHARE_WRITE;
         DWORD creation = CREATE_NEW;
         DWORD flags = FILE_ATTRIBUTE_TEMPORARY | FILE_FLAG_DELETE_ON_CLOSE;
-        hFile = CreateFileW(tmpname, access, sharemode, &sa, creation, flags, NULL);
+
+        if ((wcslen(tmppath) + wcslen(tmpname)) >= MAX_PATH)
```

Comment:
That doesn't happen for all file names. The shortest file name we'll generate is "0.tmp" and the longest is "1z141z4.tmp". So there's a possibility that some paths will work.

Either way, either we handle GetTempPathW returning too long a string before the loop or we simply `continue` here. Or both.

Reply:
Should be done now